components/reusable-components/login-form/LoginForm.tsx

```diff
@@ -0,0 +1,128 @@
+'use client';
+
+import { FormikHelpers, useFormik } from 'formik';
+import { toast } from 'react-toastify';
+import * as Yup from 'yup';
+import { useState } from 'react';
+import Turnstile from 'react-turnstile';
+import styles from './LoginForm.module.scss';
+import { useTheme } from '../../../app/context/themeContext';
+import TextInput from '../text-input/TextInput';
+import { submitLoginForm } from './SubmitLoginForm';
+
+interface FormValues {
+  email: string;
+  password: string;
+}
+const LoginForm = () => {
+  const { theme } = useTheme();
+  const isLightTheme = theme === 'light';
+
+  const [turnstileToken, setTurnstileToken] = useState<string | null>(null);
+  const initialValues = { email: '', password: '' };
+  const validationSchema = Yup.object({
+    email: Yup.string()
+      .email('*Невалидна емаил адреса')
+      .required('*Задолжително внесете емаил адреса'),
+    password: Yup.string()
+      .required('Задолжително внесете пасворд.')
+      .min(8, 'Пасвордот би требало да содржи минимум 8 знаци.')
+      .matches(
+        /^(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!%*#?&]*$/,
+        'Мора да содржи еден број и еден посебен знак'
+      ),
+  });
+  const handleSubmit = async (values: FormValues, { resetForm }: FormikHelpers<FormValues>) => {
+    if (!turnstileToken) {
+      return;
+    }
+
+    const formValues = {
+      email: values.email,
+      password: values.password,
+      cfTurnstileResponse: turnstileToken,
+    };
+
+    try {
+      const response = await submitLoginForm(formValues);
+      if (response) {
+        toast.success(response);
+
+        resetForm();
+      } else {
+        toast.error(response);
+      }
+    } catch (error) {
+      toast.error('Грешка');
+    }
+  };
+
+  const formik = useFormik({
+    initialValues,
+    validationSchema,
+    onSubmit: handleSubmit,
+  });
+
+  return (
+    <div
+      className={`${styles.loginForm} ${isLightTheme ? styles.lightLoginForm : styles.darkLoginForm}`}
+    >
+      <form className={styles.innerLoginForm} onSubmit={formik.handleSubmit}>
+        <h2 className={styles.loginTitle}>Најави се</h2>
+        <TextInput
+          placeholder="Внесете ја вашата електронска пошта"
+          label="Електронска Пошта"
+          name="email"
+          type="email"
+          field="email"
+          formik={formik}
+          isRequired
+        />
+        <TextInput
+          placeholder="Внесете го вашиот пасворд"
+          label="Пасворд"
+          name="password"
+          type="password"
```

Comment:
Why is the Login form on this PR?
This PR should only be about the shared layout the login and sign up share. Nothing else.
Additionally, I believe we already have the patterns. Aren't these the same patterns we already use in Mission and Vision?